packages/graph/hash_graph/lib/graph/src/store/postgres/query/condition.rs

```diff
@@ -0,0 +1,328 @@
+                )))),
+            )]),
+            r#"("data_types"."schema"->>'$id' = $1)"#,
+            &[&"https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1"],
+        );
+
+        test_condition(
+            &Filter::All(vec![
+                Filter::Equal(
+                    Some(FilterValue::Path(DataTypeQueryPath::BaseUri)),
+                    Some(FilterValue::Parameter(Parameter::Text(Cow::Borrowed(
+                        "https://blockprotocol.org/@blockprotocol/types/data-type/text/",
+                    )))),
+                ),
+                Filter::Equal(
+                    Some(FilterValue::Path(DataTypeQueryPath::Version)),
+                    Some(FilterValue::Parameter(Parameter::Number(1.0))),
+                ),
+            ]),
+            r#"("type_ids"."base_uri" = $1) AND ("type_ids"."version" = $2)"#,
+            &[
+                &"https://blockprotocol.org/@blockprotocol/types/data-type/text/",
+                &1.0,
+            ],
+        );
+    }
+
+    #[test]
+    fn render_any_condition() {
+        test_condition(
+            &Filter::Any(vec![Filter::Equal(
+                Some(FilterValue::Path(DataTypeQueryPath::VersionedUri)),
+                Some(FilterValue::Parameter(Parameter::Text(Cow::Borrowed(
+                    "https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1",
+                )))),
+            )]),
+            r#"("data_types"."schema"->>'$id' = $1)"#,
+            &[&"https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1"],
+        );
+
+        test_condition(
+            &Filter::Any(vec![
+                Filter::Equal(
+                    Some(FilterValue::Path(DataTypeQueryPath::BaseUri)),
+                    Some(FilterValue::Parameter(Parameter::Text(Cow::Borrowed(
+                        "https://blockprotocol.org/@blockprotocol/types/data-type/text/",
+                    )))),
+                ),
+                Filter::Equal(
+                    Some(FilterValue::Path(DataTypeQueryPath::Version)),
+                    Some(FilterValue::Parameter(Parameter::Number(1.0))),
+                ),
+            ]),
+            r#"("type_ids"."base_uri" = $1) OR ("type_ids"."version" = $2)"#,
+            &[
+                &"https://blockprotocol.org/@blockprotocol/types/data-type/text/",
+                &1.0,
+            ],
+        );
+    }
+
+    #[test]
+    fn render_not_condition() {
+        test_condition(
+            &Filter::Not(Box::new(Filter::Equal(
+                Some(FilterValue::Path(DataTypeQueryPath::VersionedUri)),
+                Some(FilterValue::Parameter(Parameter::Text(Cow::Borrowed(
+                    "https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1",
+                )))),
+            ))),
+            r#"NOT("data_types"."schema"->>'$id' = $1)"#,
+            &[&"https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1"],
+        );
+    }
```

Comment:
I think it would be nice to have a test for parameterless queries such as:
```rust
    #[test]
    fn render_without_parameters() {
        test_condition(
            &Filter::Any(vec![Filter::Equal(
                Some(FilterValue::Path(DataTypeQueryPath::Custom(Cow::Borrowed(
                    "left",
                )))),
                Some(FilterValue::Path(DataTypeQueryPath::Custom(Cow::Borrowed(
                    "right",
                )))),
            )]),
            r#"("data_types"."schema"->>'left' = "data_types"."schema"->>'right')"#,
            &[],
        );
    }
```
I can imagine that being useful for entities (may not really make sense for the ontology types)